crates/nostr/src/event/builder.rs

```diff
@@ -712,6 +712,43 @@ impl EventBuilder {
         Self::new(Kind::EventDeletion, reason.into()).tags(tags)
     }
 
+    /// Request to vanish with reason
+    ///
+    /// Must include at least one relay.
+    ///
+    /// <https://github.com/nostr-protocol/nips/blob/master/62.md>
+    pub fn request_vanish_with_reason<I, S>(relays: I, reason: S) -> Result<Self, Error>
+    where
+        I: IntoIterator<Item = RelayUrl>,
+        S: Into<String>,
+    {
+        let tags: Vec<_> = relays
+            .into_iter()
+            .map(TagStandard::Relay)
+            .map(Tag::from_standardized)
+            .collect();
+
+        if tags.is_empty() {
```

Comment:
If we go with the enum I menitoned above, this is no longer needed, otherwise the error can be added in the `Error` enum above.

Reply:
Make it as `nostr::event::tag::Error:: EmptyTags` or `nostr::event::tag::Error:: nip62(Error)`? And make the error on `nostr::nips::nip62::Error`

> It may be useful for others constructors in the future.

I think on `nostr::event::tag::Error:: EmptyTags`

Reply:
Ah no, I mean on the `nostr::event::builder::Error`, since is an error related to the event building.